smt/surrogate_models/krg_based.py

```diff
@@ -954,18 +955,21 @@ def _reduced_likelihood_function(self, theta):
             r = self._correlation_types[self.options["corr"]](theta, self.D).reshape(
                 -1, 1
             )
-        R = np.eye(self.nt) * (1.0 + nugget + noise)
+        R = np.eye(self.nt) * (1.0 + nugget)
+        R_noisy = np.eye(self.nt) * (1.0 + nugget + noise)
         R[self.ij[:, 0], self.ij[:, 1]] = r[:, 0]
```

Comment:
Why not pass `r` to _compute_sigma2_ri and set `R` only if needed ?

Reply:
fixed